### ISSUE TITLE
Todo List

### ISSUE THREAD

**Issue by Mytherin** [Mytherin/Tibialyzer#39]
I've created a wiki page for my [Tibialyzer Todo List](https://github.com/Mytherin/Tibialyzer/wiki/Todo-List). This is all the work I'm planning on doing in the (near)-future. If anyone wants to contribute and do something on the list, please feel free. If you have any suggestions for features, please post them in this topic.

**Comment by GalLuvton** [comment]
Not sure if its implemented, but:
- Set options to stop NPC popups when trading
- Change exported_log's file extention to txt on default, instead of blank

**Comment by Mytherin** [comment]
What do you mean by "Set options to stop NPC popups when trading"?

**Comment by GalLuvton** [comment]
I get that it wasn't well defined, sorry ;p
I mean, when you talk to an NPC, theres a popup about the NPC's trade list. I couldn't for the life of me find a way to disable it.

**Comment by Mytherin** [comment]
Do you mean when you look at an NPC? No popup should open when you talk to an NPC. You can disable the look popups by unchecking the "View Looked At Objects" box in the "Settings" tab.

**Comment by GalLuvton** [comment]
Well now I feel bad turning this into a spam war (I will delete all unneeded comments later).
I'm pretty sure I never looked at an NPC. But when I trade with one (could be when I buy something), I get a popup with that NPC's.
For now I can't seem to reproduce it, so I will let you know when I got the hang of what popup that really is (apparently not the NPC popup as I thought).
Please delete this comment after you read it

**Comment by Mytherin** [comment]
Three comments isn't a spam war, don't worry about it. The only oddity relating to trading with NPCs that I know of is that when you click on an item that you are buying/selling in the Trade Box, it will look at the item, which triggers Tibialyzer to open a popup of that specific item. Not a bug, but it might confuse you that Tibialyzer would open a popup in that situation.

**Comment by GalLuvton** [comment]
That is exactly the issue I was talking about!
Well that explains how to stop that from happening. Thanks!

**Comment by GalLuvton** [comment]
Is there a way to contact you regarding programming stuff? I have some pointers regarding some code segments, and they aren't really issues

**Comment by Mytherin** [comment]
Send me a message on my [Reddit Account](https://www.reddit.com/user/BadgerBadger8264/).

**Comment by GalLuvton** [comment]
Adding TODO:
- Set djinn NPC names with an extra description so you could easily tell if its green or blue djinn

**Comment by GalLuvton** [comment]
Another TODO:
- Read commands from silent-mode messages, such as "#s item@gold coin", so that the player could avoid spamming someone's private messages

**Comment by Mytherin** [comment]
@GalLuvton These messages already seem to work for me, since #s is removed before putting them in the server log.

**Comment by Fmancio** [comment]
Suggestion: a configurable health bar, maybe on top or bottom of the health and mana (curved). So honest (not botter) druids could configure it to show the knights life, making it much better to use exura sio in the middle of  the monster and players confusion.

**Comment by GalLuvton** [comment]
That would be a bit tricky.
Its not hard getting the EKs HP, but what do you expect to happeb when he goes off-screen? The bar do disappear?

**Comment by Fmancio** [comment]
The ban can turn to a different color (not used in hp or mp on tibia), with a message saying: **_[DANGER! Blocker out of the screen]_**, or similar. 
You can also color it differently from the known hp/mp colors with the direction of the blocker: north, south, east  or west (If you can also get the location of the other char). For example:               _**[Blocker lost, go north.]**_
This would be GREAT.

**Comment by Mytherin** [comment]
Already made this over the weekend, actually. See commit 7c75d87946e821ed9c7129ec7eba2fa7a745d10c, with [an image here](http://i.imgur.com/c9sCA2O.png).  I have noticed that the health is send to the client and stored in memory even if the blocker is not on the screen but relatively close (like up a floor or just out of screen). If the blocker goes way further than that the client stops receiving updates, after which point the health will not update until the blocker is found again. I don't think it's a very big deal anyway because when the blocker is far away you can't heal him anyway.

**Comment by Fmancio** [comment]
=D
Man, you are awesome. When this feature can be enabled, maybe next beta?
Thanks for tibialyzer, me and my friends are in love with it lol.

**Comment by Mytherin** [comment]
I will update the beta release later today.

**Comment by bneedy** [comment]
I know you have a lot of stuff you are wanting to implement, but one thing I think would be cool would be a notification for when someone on your vip list logged in. Just an idea.

**Comment by Gaabe** [comment]
I want to get screenshots only from decent loots, above a certain value, but I still want it to show the loot notification for loot with lower value. Really looking forward to this feature, two separate loot value tresholds, one for display and another for screenshot.

**Comment by KuerbisGnu** [comment]
- [ ] Update "Search Commands" in Help Tab

**Comment by mathiasbynens** [comment]
I’d love to see an entry in the “Summary” view that displays how much experience is still needed to level up **OR** how much longer it takes to level-up based on the current xp/h.

**Comment by Wofrado** [comment]
I was wondering if we could have different lists within Health List?

**Comment by bysiek** [comment]
Hey, just came up an idea to scan log and give out output of mini world chagnes, i often go to Adventurers Guild several times daily to check out whether i remember which are active, and which are not.
mwc@, or miniworldchange@ that would give window with i.e:
"Yasir,
Nightmare Isles - Drefia, 
Spirit Gate - Ghostlands..."
etc, shouldn't be hard to add, but somewhat useful.
Regards,

**Comment by benjaminkomen** [comment]
Hey Mytherin,
It would be a really useful feature for TibiaWiki, to have a list of creature, items, NPCs which Tibialyzer users encounter but are not in the database. So we can add them to TibiaWiki.

**Comment by Mytherin** [comment]
@benjaminkomen 
I'm currently not gathering data from players, and I'm currently not planning on adding that. One solution would be for me to log unknown creatures/items as errors and then players could manually send you the error logs, but I'm not sure how useful that would be for you.

**Comment by benjaminkomen** [comment]
Yes, those error logs would be extremely useful. Maybe the easiest implementation is a button in Tibialyzer which triggers sending an e-mail with the error log?

**Comment by bneedy** [comment]
I think it would be awesome if the player window showed if the user was online or not.  I don't understand fully how you are gathering that data from the web but it seems like it should be possible since there is a status "online" field.

**Comment by kugrjado** [comment]
I have actually two suggestions to make implementing of which would make more visibility on how effective the hunt is.
Add extra time-stamps to hunt log saved in Tibialyzer.

How it works now:

1. You kill a creature and you get a loot notification with the time, let's say 11:01 creature killed: some loot.
2. You kill another creature 5 secs after than and another creature 5 secs after that.
3. All loot notifications say you killed them at 11:01, which is true obviously.

BUT! How it could actually work and what would be the possible gain if:

1. Tibialyzer registers the loot notification with additional time stamp so instead of 11:01 it would look like 11:01:00 and with next creatures 11:01:05, 11:01:10, right?
2. In summary window there would be extra information on how many specific creatures we kill per minute, and also how much money we make per minute.

Now you may wonder why extra time stamp, if you can make it worth without knowing the seconds? Well...

3. After clicking on a "creature per minute" or "money per minute" information we see a new window with a linear graph! On this graph we can see the variation of how effective we are on our hunt.

Sometimes many people waste time on killing creatures in some area of the hunting location that if they would skip these creatures and move futher they would profit more (exp and money).
